File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0, right = heights.size() - 1;
        int max_water = INT_MIN;
        while (left < right) {
            int water = (right-left) * min(heights[left], heights[right]);
            max_water = max(max_water, water);
            (heights[left] > heights[right]) ? right-- : left++;
        }
        return max_water;
    }
};

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        n = len(heights)
        maxArea = 0
        stack = []
        for i, v in enumerate(heights):
            start = i
            while stack and stack[-1][1] > v:
                topIdx,topVal = stack.pop()
                area = topVal * (i - topIdx)
                maxArea = max(maxArea, area)
                start = topIdx
            stack.append([start,v])
        while stack:
            topIdx,topVal = stack.pop()
            area = topVal * (n - topIdx)
            maxArea = max(maxArea, area)
        return maxArea